### DISCUSSION TITLE
how to conditional compile App.axml per platform

### DISCUSSION THREAD

**Discussion by CastelloBrancoTecnologia** [AvaloniaUI/Avalonia#14128] [Q&A] [answered]
I have two App.axaml files pointing to same App.cs 

one is App-All.axaml and other is App-Browser.axaml

i What in shared project make some like this

```
	<Choose>
		<When Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'"  >
		
			<ItemGroup>
				<None Remove="App-Browser.*" />
			</ItemGroup>
		</When>
		<Otherwise>
			<ItemGroup>
				<None Remove="App-All.*" />
			</ItemGroup>
		</Otherwise>
	</Choose> 
```

How can i do this ?

				`<None Remove="App-All.axaml" />`???

 is not working

i am gething 
Erro	AVLN:0002	Duplicate x:Class directive, SecoviRioApp.App is already used in /App-All.axaml	SecoviRioApp	C:\Users\cesar\Documents\GitHub\SecoviRioAppNovo\SecoviRioApp\App-Browser.axaml	1

**Comment by CastelloBrancoTecnologia** [comment]
Also i want to make this with other Pages.XAML 

to have Pages Per plataform

**Comment by workgroupengineering** [comment, marked as answer]
```xml
<Choose>
  <When Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios' or $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'"  >
  <ItemGroup>
    <AvaloniaXaml Remove="App-Browser.axaml" />
  </ItemGroup>
  </When>
  <Otherwise>
    <ItemGroup>
        <AvaloniaXaml Remove="App-All.axaml" />
    </ItemGroup>
  </Otherwise>
</Choose> 
```

**Reply by maxkatz6** [reply]
Aside from "please use OnPlatform instead" in this case I would recommend using "net8.0-browser" target framework.
So it can be simplified to `$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'browser'` condition.

**Comment by thevortexcloud** [comment]
Honestly you are better off just breaking it up into multiple projects with a shared project containing all your views. Which is how the default template works.

**Reply by cesarchefinho** [reply]
it is already this way, a shared project with all views, but i need to not include in app.axaml material.avalonia styles because it is buggy when platform is browser and browser diaplay none with matetial styles.

it is only a work around to have two app.axaml style, one for all plataforms and one for browser plataform until material.avalonia fixes it

**Reply by maxkatz6** [reply]
https://docs.avaloniaui.net/docs/guides/platforms/platform-specific-code/xaml#xml-combining-syntax

**Comment by maxkatz6** [comment]
Please use conditional XAML, specifically OnPlatform markup extension instead of duplicating XAML files:
https://docs.avaloniaui.net/docs/guides/platforms/platform-specific-code/xaml#xml-combining-syntax

Just added that styles example.
